Source/Common/GLTFUnarchiver.swift

```diff
@@ -794,7 +800,11 @@ public class GLTFUnarchiver {
                 image = try loadImageData(from: data)
             } else {
                 let url = URL(fileURLWithPath: uri, relativeTo: self.directoryPath)
-                image = try loadImageFile(from: url)
+                // Even if we shouldn't embed external images, try loading the image file for now
+                // TODO: figure out how to check if `url` exists on-disk rather than using `loadImageFile()` to do this
+                let loadedImage = try loadImageFile(from: url)
```

Comment:
This is the most awkward part of this revision. As indicated in the comments, I’m essentially using `loadImageFile()` as a test to see if the file exists on-disk. There are a few ways to improve this:

- Add a helper function to simply check if the image URL exists on-disk when `self.embedExternalImages` is `false`
- Call `loadImageFile()` only when `self.embedExternalImages` is `true`